packages/noya-state/src/reducers/application.ts

```diff
@@ -845,6 +866,46 @@ export function reducer(
         });
       });
     }
+    case 'setLayerStyle': {
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      const [, id, style] = action;
+
+      return produce(state, (state) => {
+        accessPageLayers(state, pageIndex, layerIndexPaths).forEach((layer) => {
+          if (!layer.style) return;
+
+          if (style) {
+            layer.sharedStyleID = id;
+            layer.style = produce(style, (style) => {
+              style.do_objectID = uuid();
+              return style;
+            });
+          } else {
+            layer.sharedStyleID = undefined;
+          }
+        });
+      });
+    }
+    case 'updateLayerStyle': {
+      const [, id, style] = action;
+
+      return produce(state, (state) => {
+        if (!style) return;
+
+        const sharedStyle = state.sketch.document.layerStyles.objects;
```

Comment:
This can be `sharedStyle` => `sharedStyles`

So below we can have `SStyle` => `sharedStyle`